pkg/backend/display/diff.go

```diff
@@ -483,3 +484,52 @@ func renderDiffResourceOutputsEvent(
 	}
 	return out.String()
 }
+
+// CreateDiff renders a view of the given events, enforcing an order of rendering that is consistent
+// with the diff view.
+func CreateDiff(events []engine.Event, displayOpts Options) (string, error) {
+	buff := &bytes.Buffer{}
+
+	seen := make(map[resource.URN]engine.StepEventMetadata)
+	displayOpts.SummaryDiff = true
+
+	if displayOpts.Color == "" {
+		// ColorizeWithMaxWidth panics if the colorization mode is not recognized, so we enforce it here.
+		return "", errors.New("color must be specified")
+	}
```

Comment:
should we set a reasonable default instead of erroring?

Reply:
+1. This is breaking change. The function did not return error before, why is this necessary now?

Reply:
It panicked before, actually. This protects against a panic in the server side code. I was unsure of choosing a default here where none existed before, but couldn't find any clear evidence either way.

Reply:
Panic on line 48, discovered while writing tests:

https://github.com/pulumi/pulumi/blob/e03d6f2c4cd1a282b6f6f2be18606c129ec02622/sdk/go/common/diag/colors/diag.go#L37-L48

Reply:
Ah, I see. In that case, relaxing it is not needed, I withdraw my suggestion.

Reply:
I think it's reasonable to treat `Never` as the default here. Though arguably a `panic` is the right thing inasmuch as some choice for colorization is a precondition?

Reply:
(regardless, I don't think that this function should return an error--if a colorization is not specified, that's a programming error, not a runtime error)

Reply:
Aren't panics more expensive (and considered P0s/P1s?) - I would prefer to have a "safe but non-panicking" API that aborts early than one that conditionally panicks on a data-dependent path that could change in the future.

Reply:
@pgavlin if you're OK with it, maybe we default to colorization.Never (which produces "clean" text output).

I could go either way on keeping the return type `(string, error)`, in favor: forward API compatibility in case some other rendering option is fallible and bubbles up an error, against: it reinforces that this method _should_ be infallible.

Reply:
> Aren't panics more expensive (and considered P0s/P1s?) - I would prefer to have a "safe but non-panicking" API that aborts early than one that conditionally panicks on a data-dependent path that could change in the future

FWIW if we were working in a language with enums, the colorization param would be of an enum type, which is what is is meant to be semantically. So I think panicking is in line with those semantics (and in general has been the philosophy throughout the product).

> if you're OK with it, maybe we default to colorization.Never (which produces "clean" text output).

Yeah I think that this is a fine approach. I think that removing the error parameter and using that default or removing the error and retaining the panic (or adding a `contract.Require` or something) would be great.